### PR TITLE
LPS-80511 Portlet 3.0: Implement javax.portlet.container-runtime-option in PortletTracker

### PR DESCRIPTION
**Pull request by ngriffin7a**

### CODE REVIEW

**Comment by ngriffin7a** [comment]
ci:test

**Comment by ngriffin7a** [comment]
ci:retest

**Comment by ngriffin7a** [comment]
ci:retest

**Comment by ngriffin7a** [comment]
ci:reopen

**Comment by ngriffin7a** [comment]
@shuyangzhou This PR is simply collecting some container-runtime-option element info from portlet.xml and therefore has a very minimal impact on things. Additionally, there is only "1 failure unique to this pull" which is also found in https://github.com/shuyangzhou/liferay-portal/pull/6415 so it must be unrelated.

Please review when you have time. Thanks.

File: modules/apps/static/portal-osgi-web/portal-osgi-web-portlet-tracker/src/main/java/com/liferay/portal/osgi/web/portlet/tracker/internal/PortletTracker.java [hunk 1]
```diff
@@ -427,6 +429,69 @@ protected void collectCacheScope(
...
+		PortletApp portletApp = portletModel.getPortletApp();
+
+		Map<String, String[]> containerRuntimeOptionsMap =
+			portletApp.getContainerRuntimeOptions();
+
+		List<String> containerRuntimeOptions = StringPlus.asList(
+			serviceReference.getProperty(
+				"javax.portlet.container-runtime-option"));
+
+		for (String containerRuntimeOption : containerRuntimeOptions) {
```
Comment (shuyangzhou) [inline modules/apps/static/portal-osgi-web/portal-osgi-web-portlet-tracker/src/main/java/com/liferay/portal/osgi/web/portlet/tracker/internal/PortletTracker.java]: This for loop and https://github.com/shuyangzhou/liferay-portal/pull/6408/commits/e1e9ab78e6de30320faecd7f39a44dd498531f47#diff-28e3cf259d89b86f287fb1efd5d889dfR457 seem very likely can be merged to avoid doubling the key/value split. If so please resend this with them merged. Thanks!

Comment (ngriffin7a) [reply modules/apps/static/portal-osgi-web/portal-osgi-web-portlet-tracker/src/main/java/com/liferay/portal/osgi/web/portlet/tracker/internal/PortletTracker.java]: @shuyangzhou I looked at the code and I don't think that the two for-loops can be merged easily.

**Background**:

The XSD for container-runtime-option allows for multiple values:
```
	<complexType name="container-runtime-optionType">
		<annotation>
			<documentation>...</documentation>
		</annotation>
		<sequence>
			<element name="name" type="portlet:nameType"/>
			<element name="value" type="portlet:valueType" minOccurs="0" maxOccurs="unbounded"/>
		</sequence>
	</complexType>
```

Footnote "2" of the [https://dev.liferay.com/develop/reference/-/knowledge_base/7-0/portlet-descriptor-to-osgi-service-property-map](https://dev.liferay.com/develop/reference/-/knowledge_base/7-0/portlet-descriptor-to-osgi-service-property-map) document indicates that the way to specify multiple values is to specify the property more than once:
> [2] Multiples of these properties may be used. This results in an array of values.

This approach is documented in https://issues.liferay.com/browse/LRDOCS-5269.

In other words, the "javax.portlet.container-runtime-option" property can be specified multiple times in order to accommodate multiple values. For example:
```
@Component(
	immediate = true,
	property = {
		"javax.portlet.container-runtime-option=foo;1",
		"javax.portlet.container-runtime-option=foo;2",
		"javax.portlet.container-runtime-option=foo;3",
		"javax.portlet.version=3.0"
	},
	service = Portlet.class
)
public class TestPortlet {
}
```

The **first** for-loop removes any values from the map.

The **second** for-loop iterates over all the entries in the "javax.portlet.container-runtime-option" array. Each iteration of the loop would add a value to the underlying array of values in the map.

File: modules/apps/static/portal-osgi-web/portal-osgi-web-portlet-tracker/src/main/java/com/liferay/portal/osgi/web/portlet/tracker/internal/PortletTracker.java [hunk 2]
```diff
@@ -427,6 +429,69 @@ protected void collectCacheScope(
...
+				containerRuntimeOption, CharPool.SEMICOLON);
+
+			if (parts.length == 2) {
+				name = parts[0];
+				value = parts[1];
+			}
+
+			String[] values = containerRuntimeOptionsMap.get(name);
+
+			if (values == null) {
```
Comment (shuyangzhou) [inline modules/apps/static/portal-osgi-web/portal-osgi-web-portlet-tracker/src/main/java/com/liferay/portal/osgi/web/portlet/tracker/internal/PortletTracker.java]: @ngriffin7a can this "values" not be null? Did not we just remove the exact same name from the containerRuntimeOptionsMap in the first loop?
The later put here https://github.com/shuyangzhou/liferay-portal/pull/6408/commits/e1e9ab78e6de30320faecd7f39a44dd498531f47#diff-28e3cf259d89b86f287fb1efd5d889dfR490 is under a different name, on the next iteration it still won't see the "values" under "name", so the 2nd iteration would just throw away 1st iteration's "values", right?

**Comment by ngriffin7a** [comment]
See https://github.com/shuyangzhou/liferay-portal/pull/6431
